src/views/catalog/CreateFromInstanceTypes/components/SelectInstanceTypeSection/components/CategoryCard/components/InstanceTypesMenu/EnhancedMenuToggle.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { ElementType, RefObject, SetStateAction } from 'react';
+
+import CustomToggle from '@catalog/CreateFromInstanceTypes/components/SelectInstanceTypeSection/components/CategoryCard/components/InstanceTypesMenu/CustomToggle';
+import {
+  CLICK,
+  ESCAPE,
+  KEYDOWN,
+  TAB,
+} from '@catalog/CreateFromInstanceTypes/components/SelectInstanceTypeSection/utils/constants';
+
+type EnhancedMenuToggleProps = {
+  isOpen: boolean;
+  menuRef: RefObject<HTMLElement>;
+  onToggleClick: (state: SetStateAction<boolean>) => void;
+  toggleComponent: ElementType;
+  toggleRef: RefObject<HTMLButtonElement>;
+  customData?: any;
```

Comment:
any?

Reply:
This is also a part of keeping the menu components more general. The menu components don't need to know what type of data the caller and eventual child menu and toggle components need.